wallet-new/test/unit/UTxO/Interpreter.hs

```diff
@@ -404,87 +524,65 @@ instance DSL.Hash h Addr => Interpret h (DSL.Utxo h Addr) where
+        pm <- asks magic
         case classifyInputs inps of
           Left err ->
             throwError err
-          Right (InputsRegular inps') -> withProtocolMagic $ \pm -> withConfig $
+          Right (InputsRegular inps') -> withConfig $
             return . either absurd identity $
               makeMPubKeyTx
                 pm
                 (Right . FakeSigner . regKpSec)
                 (NE.fromList inps')
                 (NE.fromList outs)
-          Right (InputsRedeem (kp, inp)) -> withProtocolMagic $ \pm -> withConfig $
+          Right (InputsRedeem (kp, inp)) -> withConfig $
             return $
               makeRedemptionTx
                 pm
                 (redKpSec kp)
                 (NE.fromList [inp])
                 (NE.fromList outs)
 
--- | Interpretation of a list of transactions, oldest first
---
--- Each transaction becomes part of the context for the next.
+-- | Interpretation of a block of transactions, oldest first
 instance DSL.Hash h Addr => Interpret h (DSL.Block h Addr) where
-  type Interpreted (DSL.Block h Addr) = RawResolvedBlock
+  -- The block and the EBB, if any
+  type Interpreted (DSL.Block h Addr) = (RawResolvedBlock, Maybe GenesisBlock)
 
-  int :: forall e m. (Monad m)
-      => DSL.Block h Addr -> IntT h e m RawResolvedBlock
+  int :: forall e m. Monad m
+      => DSL.Block h Addr -> IntT h e m (RawResolvedBlock, Maybe GenesisBlock)
   int (OldestFirst txs) = do
-      -- Create an epoch boundary block on the epoch boundary
-      ic <- get
-      mebb <- if isEpochBoundary ic
-              then do
-                (pc, slots) <- liftTranslateInt $ do
-                  cc <- asks tcCardano
-                  return ( genesisProtocolConstantsToProtocolConstants . gdProtocolConsts $ ccData cc
-                         , ccEpochSlots cc
-                         )
-                let newLeaders = give pc $ followTheSatoshi slots boringSharedSeed (HM.toList $ icCrucialStakes ic)
-                    newEpoch = nextEpoch $ icEpoch ic
-                    sbb = mkGenesisBlock testProtocolMagic (Right $ icPrevBlock ic) newEpoch newLeaders
-                pushEpochBoundary sbb >> return (Just sbb)
-              else return Nothing
-
       (txs', resolvedTxInputs) <- unpack <$> mapM int txs
-      leaders <- gets icEpochLeaders
-      prev    <- gets icPrevBlock
-      slot    <- gets icNextSlot
-      block   <- liftTranslateInt $ mkBlock leaders prev slot txs'
-      pushBlock block
-      return $ mkRawResolvedBlock block mebb resolvedTxInputs
+      pushCheckpoint $ \prev slot -> do
+        pc    <- asks constants
+        block <- mkBlock
+                   (icEpochLeaders prev)
+                   (icBlockHeader  prev)
+                   slot
+                   txs'
+        let raw = mkRawResolvedBlock block resolvedTxInputs
+        checkpoint <- mkCheckpoint prev slot raw
+        if siSlot slot == localSlotIndexMaxBound pc
```

Comment:
[minor] I find this slightly more confusing, since it's not obvious what `localSlotIndexMaxBound` is and why we should care here.

Reply:
Reintroduced

````haskell
isEpochBoundary :: ProtocolConstants -> SlotId -> Bool
isEpochBoundary pc slot = siSlot slot == localSlotIndexMaxBound pc
```

It's removal was an unintended consequence of the refactoring.